File: sdkaodocsjs/java/discoveryservice.java
/*
*
* AODocs SDK Version 1.0.1.alpha
* Copyright (c) 2014 Dedalus Prime
* 
* Created by Wilson.Souza
* Date 10-03-2014
* 
* Description: offer support for called to AODocs Web Service by java script
* 
*/
function DiscoveryService()
{
   this.szHttpAddress = "https://api-dot-ao-docs.appspot.com/_ah/api/discovery/v1/apis/";
   
   /* 
    * first method retrieve the description of a particular version of an api 
    * call GET rest api
    * */
   this.GetRest = function(szAPIName, szVersion)
   {
      var szCmd = this.szHttpAddress + szAPIName + "/" + szVersion + "/rest"; 
      /* no source code */
      return {Result: "command without effect!"};
   };
   
   /*
    * retrieve the list of APIS supported at this end point
    */
   this.List = function(szAPIName, bPreferred)
   {
      var szCmd = this.szHttpAddress + "?name=" + szAPIName + "&preferred=" + bPreferred;
      /* no source code */
      return {Result: "command without effect!"};
   };
}

File: sdkaodocsjs/js/discoveryservice.js
/*
*
* AODocs SDK Version 1.0.1.alpha
* Copyright (c) 2014 Dedalus Prime
* 
* Created by Wilson.Souza
* Date 10-03-2014
* 
* Description: offer support for called to AODocs Web Service by java script
* 
*/
document.writeln( "<script type='text/javascript' scr='dedalus.js'></script>" );
//-----------------------------------------------------------------------------------------------//
function DiscoveryService( hAPI, szVersion )
{
   this.m_api = hAPI;
   this.m_version = szVersion;
   this.m_url = 'https://api-dot-ao-docs.appspot.com/_ah/api/discovery/v1';
}
//-----------------------------------------------------------------------------------------------//
DiscoveryService.prototype.getRest = function(Error)
{
   var url = [this.m_url, '/apis/', this.m_api, '/', this.m_version, '/rest'].join( '' );
   var result = { 'result': null, 'success': false };

   $.ajax(
      {
         type: 'GET',
         url: url,
         data: {},
         success: function ( data )
         {
            result.result = data;
            result.success = !result.success;
         },
         error: function ( e )
         {
            result.result = e;
            Error( JSON.parse( e.responseText ) );
         },
         dataType: 'json',
         async: true
      } );
   return result;
}
//-----------------------------------------------------------------------------------------------//
DiscoveryService.prototype.list = function ( szName, bpreferred, Error )
{
   var url = [this.m_url, '/apis'].join( '' );
   var params =
      {
         'name': szName,
         'preferred': bpreferred
      };
   var result = { 'result': null, 'success': false };

   $.ajax(
      {
         type: 'GET',
         url: url + '?' + CreateParams( params ),
         data: {},
         success: function ( data )
         {
            result.result = data;
            result.success = !result.success;
         },
         error: function ( e )
         {
            result.result = e;
            Error( JSON.parse( e.responseText ) );
         },
         dataType: 'json',
         async: true
      } );
   return result;
}
//-----------------------------------------------------------------------------------------------//
